src/main/java/cz/cvut/kbss/termit/service/business/SparqlExternalVocabularyService.java

```diff
@@ -0,0 +1,223 @@
+import java.util.HashMap;
+import java.util.List;
+import org.eclipse.rdf4j.query.BindingSet;
+import org.eclipse.rdf4j.query.GraphQuery;
+import org.eclipse.rdf4j.query.GraphQueryResult;
+import org.eclipse.rdf4j.query.QueryEvaluationException;
+import org.eclipse.rdf4j.query.TupleQuery;
+import org.eclipse.rdf4j.query.TupleQueryResult;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.eclipse.rdf4j.repository.RepositoryException;
+import org.eclipse.rdf4j.repository.sparql.SPARQLRepository;
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.rio.Rio;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Business logic concerning external vocabularies imported from SPARQL repository.
+ */
+@Service
+public class SparqlExternalVocabularyService implements ExternalVocabularyService{
+    
+    private static final Logger LOG = LoggerFactory.getLogger(SparqlExternalVocabularyService.class);
+    
+    private final Configuration config;
+    private final VocabularyRepositoryService repositoryService;
+    
+    private final AccessControlListService aclService;
+    private ApplicationEventPublisher eventPublisher;
+    
+    private static final String LIST_AVAILABLE_VOCABULARIES_QUERY = "import/listAvailableVocabularies.rq";
+    private static final String EXPORT_FULL_VOCABULARY_QUERY = "import/exportFullVocabulary.rq";
+    
+        public SparqlExternalVocabularyService(VocabularyRepositoryService repositoryService,
+                             AccessControlListService aclService,
+                             Configuration config) {
+        this.repositoryService = repositoryService;
+        this.aclService = aclService;
+        this.config = config;
+    }
+    
+    
+
+/**
+ * Sends a SPARQL query to fetch list of available vocabularies.
+ * 
+ * @return list of available vocabulary information or null if connection failed
+ */
+    @Override
+    public List<RdfsResource> getAvailableVocabularies() {
+
+        List<RdfsResource> response;
+        try {
+            SPARQLRepository sparqlRepo = initSparqlRepository();
+
+            try (RepositoryConnection conn = sparqlRepo.getConnection()) {
+                String sparqlQuery = Utils.loadQuery(LIST_AVAILABLE_VOCABULARIES_QUERY);
+                TupleQuery query = conn.prepareTupleQuery(sparqlQuery);
+
+                TupleQueryResult result = query.evaluate();
+                response = extractListOfAvailableVocabularies(result);
+
+            } catch (QueryEvaluationException e) {
+                LOG.error(e.getMessage());
+                return List.of();
+            }
+            sparqlRepo.shutDown();
```

Comment:
How do you shut down the repo, when the catch above already returns. This should be in a finally block at least.